Applications/spire/source/time_and_sales/time_and_sales_table_view.cpp

```diff
@@ -130,6 +130,20 @@ time_and_sales_table_view::time_and_sales_table_view(QWidget* parent)
 
 void time_and_sales_table_view::set_model(time_and_sales_window_model* model) {
   m_model = model;
+  m_transition_widget.reset();
+  m_loading_widget.reset();
+  if(m_model->is_loading()) {
+    QTimer::singleShot(1000, [=] { show_transition_widget(); });
```

Comment:
Never use singleShot without specifying a clear owner, otherwise this will crash if the window closes before the 1 second timeout.